runtime/drivers/duckdb/olap.go

```diff
@@ -181,6 +186,344 @@ func (c *connection) EstimateSize() (int64, bool) {
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("DROP TABLE IF EXISTS %s", safeSQLName(name)),
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+
+	// ignore query cancellations
+	ctx = context.WithValue(context.Background(), connCtxKey{}, connFromContext(ctx))
+	sourceDir := filepath.Join(filepath.Dir(c.config.DBFilePath), name)
+	version, exist, err := c.tableVersion(name)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		return nil
+	}
+	// drop view
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("DROP VIEW IF EXISTS %s", safeSQLName(name)),
+		Priority:    100,
+		LongRunning: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	oldDB := dbName(name, version)
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("DETACH %s", safeSQLName(oldDB)),
+		Priority:    100,
+		LongRunning: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// delete source directory
+	return os.RemoveAll(sourceDir)
+}
+
+// InsertTableAsSelect implements drivers.OLAPStore.
+func (c *connection) InsertTableAsSelect(ctx context.Context, name string, byName bool, sql string) error {
+	var insertByNameClause string
+	if byName {
+		insertByNameClause = "BY NAME"
+	} else {
+		insertByNameClause = ""
+	}
+
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("INSERT INTO %s %s (%s)", safeSQLName(name), insertByNameClause, sql),
+			Priority:    1,
+			LongRunning: true,
+		})
+	}
+	version, exist, err := c.tableVersion(name)
+	if err != nil {
+		return err
+	}
+	if !exist {
+		return fmt.Errorf("table %q does not exist", name)
+	}
+	return c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("INSERT INTO %s.default %s (%s)", safeSQLName(dbName(name, version)), insertByNameClause, sql),
+		Priority:    1,
+		LongRunning: true,
+	})
+}
+
+// RenameTable implements drivers.OLAPStore.
+func (c *connection) RenameTable(ctx context.Context, oldName, newName string, view bool) error {
+	if oldName == newName {
+		// todo check case sensitivity
+		return fmt.Errorf("old and new name are same")
+	}
+	if view {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("ALTER VIEW %s RENAME TO %s", safeSQLName(oldName), safeSQLName(newName)),
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("ALTER TABLE %s RENAME TO %s", safeSQLName(oldName), safeSQLName(newName)),
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+
+	// ignore query cancellations
+	ctx = context.WithValue(context.Background(), connCtxKey{}, connFromContext(ctx))
+	newSrcDir := filepath.Join(filepath.Dir(c.config.DBFilePath), newName)
+	// new name should not exist
+	exist := dirExist(newSrcDir)
+	if exist {
+		return fmt.Errorf("table %q already exist", newName)
+	}
```

Comment:
I don't think it should fail here – don't renames need to be implemented as force renames? Otherwise, you would need to drop the previous table before renaming, causing a period of failing queries

Reply:
There is no force rename in duckdb SQL as well. Do we want this functionality to be different than duckdb ?

Reply:
Also even in the current logic we are closing all running queries then dropping original table and renaming temporary table to the new table.

Reply:
There is a force rename in DuckDB:
```sql
BEGIN:
DROP TABLE IF EXISTS newName;
ALTER TABLE name RENAME TO newName;
COMMIT;
```
Which is how the current logic in reconcile works (though using a simulated transaction by passing `tx=true` to `WithConnection`).